AndelaOSP/art-backend
# #165376778 Add fields to user and asset models.

## What does this PR do?
This PR adds department fields to user and asset tables. It also adds active and paid_postpaid fields to the asset table as well. A table is also added to track owners of assets, different from people to whom assets are assigned. Third, the PR restricts users to only view assets in their departments and location

## Description of Task to be completed?
Currently, tracking of which departments users and assets belong to is not being done. There is also no tracking of actual owners (department, user or workspace) of an asset but only an assignee. Thirdly, users can view assets irrespective of which departments or locations they belong. The changes in this PR address these issues.

## How should this be manually tested?
1. On the browser, go to URL `http://127.0.0.1:8000/api/v1/users/` and add a user with an additional key-value for `department` for an existent department. A response with the given department should be returned

2. Go to `http://127.0.0.1:8000/api/v1/assets/` and post an asset with additional key-value pairs for `department`, `active` and `prepaid_postpaid`. A response with the same data should be returned.

3. If there are any existing assets not in the same department or location as a user interacting with `http://127.0.0.1:8000/api/v1/assets/` using a get method, these assets will not appear in the response.

4. Go to `http://127.0.0.1:8000/api/v1/manage-assets/{uuid}/` and patch an asset's active, prepaid_postpaid or department fields. A response with the updated asset field should be returned. This is dependent on which type of asset is being worked on, with only simcards and mifis allowed to have their active and prepaid_postpaid fields.

5. Endpoints that were previously using current_owner/previous_owner are now using current_assignee/previous_assignee.

## What are the relevant pivotal tracker stories?
[#165376778](https://www.pivotaltracker.com/story/show/165376778)
